pkg/action/archive.go

```diff
@@ -85,512 +72,22 @@ func getExt(path string) string {
-				return fmt.Errorf("failed to create directory: %w", err)
-			}
-		case tar.TypeReg:
-			if err := os.MkdirAll(filepath.Dir(target), 0o700); err != nil {
-				return fmt.Errorf("failed to create parent directory: %w", err)
-			}
-
-			// #nosec G115
-			out, err := os.OpenFile(target, os.O_RDWR|os.O_CREATE|os.O_TRUNC, os.FileMode(header.Mode))
-			if err != nil {
-				return fmt.Errorf("failed to create file: %w", err)
-			}
-
-			if _, err := io.Copy(out, io.LimitReader(df.Data, maxBytes)); err != nil {
-				out.Close()
-				return fmt.Errorf("failed to copy file: %w", err)
-			}
-
-			if err := out.Close(); err != nil {
-				return fmt.Errorf("failed to close file: %w", err)
-			}
-		case tar.TypeSymlink:
-			// Skip symlinks for targets that do not exist
-			_, err = os.Readlink(target)
-			if os.IsNotExist(err) {
-				continue
-			}
-			// Ensure that symlinks are not relative path traversals
-			// #nosec G305 // L208 handles the check
-			linkReal, err := filepath.EvalSymlinks(filepath.Join(d, header.Linkname))
-			if err != nil {
-				return fmt.Errorf("failed to evaluate symlink: %w", err)
-			}
-			if !isValidPath(linkReal, d) {
-				return fmt.Errorf("symlink points outside temporary directory: %s", linkReal)
-			}
-			if err := os.Symlink(linkReal, target); err != nil {
-				return fmt.Errorf("failed to create symlink: %w", err)
-			}
-		}
-	}
-
-	return nil
-}
-
-func extractBz2(ctx context.Context, d, f string) error {
-	logger := clog.FromContext(ctx).With("dir", d, "file", f)
-	logger.Debug("extracting bzip2 file")
-
-	// Check if the file is valid
-	_, err := os.Stat(f)
-	if err != nil {
-		return fmt.Errorf("failed to stat file: %w", err)
-	}
-
-	tf, err := os.Open(f)
-	if err != nil {
-		return fmt.Errorf("failed to open file: %w", err)
-	}
-	defer tf.Close()
-	// Set offset to the file origin regardless of type
-	_, err = tf.Seek(0, io.SeekStart)
-	if err != nil {
-		return fmt.Errorf("failed to seek to start: %w", err)
-	}
-
-	br := bzip2.NewReader(tf)
-	uncompressed := strings.TrimSuffix(filepath.Base(f), ".bz2")
-	uncompressed = strings.TrimSuffix(uncompressed, ".bzip2")
-	target := filepath.Join(d, uncompressed)
-	if err := os.MkdirAll(d, 0o700); err != nil {
-		return fmt.Errorf("failed to create directory for file: %w", err)
-	}
-
-	// #nosec G115 // ignore Type conversion which leads to integer overflow
-	// header.Mode is int64 and FileMode is uint32
-	out, err := os.OpenFile(target, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0o600)
-	if err != nil {
-		return fmt.Errorf("failed to create file: %w", err)
-	}
-	defer out.Close()
-	if _, err := io.Copy(out, io.LimitReader(br, maxBytes)); err != nil {
-		out.Close()
-		return fmt.Errorf("failed to copy file: %w", err)
-	}
-	return nil
-}
-
 func extractNestedArchive(
 	ctx context.Context,
 	d string,
 	f string,
 	extracted *sync.Map,
 ) error {
 	isArchive := false
-	ext := getExt(f)
-	if _, ok := archiveMap[ext]; ok {
+	// zlib-compressed files are also archives
```

Comment:
These lines, L101-112, L152-162, and the additions in `zlib.go` are the only material changes in this PR. The rest are just moving existing code into new files (and updating the debug logging in `gzip.go`.